JavaScript/es6Guidelines.md

```diff
@@ -0,0 +1,129 @@
+
+## General Guidelines
+
+### CONSTs and LETs over VARs
+
+Recommended article about var/let/const - [erric-elliot/javascript-es6-var-let-or-const](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75#.uw6acfhkw)[.](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75#.uw6acfhkw)
+
+We recommend following rule: const > let > var.  
+We listed some useful tips to help you decide which declaration to use.
+
+1.  **Are you going to reassign the variable's value?**  
+    **no)** use const  
+    **yes)** use let
+
+    _"No reassign" means - you are not going to change the reference._  
+    
+    **Something like that is correct:**
+
+    ```javascript
+    const point = {
+        x: 5,
+        y: 5
+    };
+    
+    point.x = 10
+    ```
+
+   **However, something like this is not:**
+
+    ```javascript
+    const shape = 'triangle';
+    shape = 'circle';
+  ```
+
+2.  **Always use one let/const/var declaration in block**  
+    You should put lets and consts on top of the block.  
+    Even though consts and lets are not hoisted like vars it's a good practice to put those declarations on top of the block to increase code readability.  
+    
+3.  **Try to keep let/const/var scope as small possible**
+
+#### When to use VARs?
+
+**Never**. No, seriously. There are almost no valid uses for `var` right now.
+
+### Default Exports
+
+**Always add export default**

```

Comment:
I think it is a good practice to think if my module should have a default export. That should lead us to create more smaller modules and promote composability over large modules with many exports.